artiq/coredevice/comm_kernel.py

```diff
@@ -520,10 +531,24 @@ def service(obj, attr, value): return setattr(obj, attr, value)
             logger.debug("rpc service: %d %r %r = %r",
                          service_id, args, kwargs, result)
 
+            original_send = self._write
+            payload = bytearray()
+
+            def cache_write(data):
+                nonlocal payload, original_send
+                payload += data
+                # send if the payload is already pretty big
+                if len(payload) > 8192:
+                    original_send(payload)
+                    payload.clear()
+            self._write = cache_write
```

Comment:
Can this logic be implemented in the ``_write`` method instead of replacing it? (and add a ``_flush`` method and/or something to enable/disable the cache I suppose)

Reply:
Yes, but I was a bit afraid of some edge cases which may cause the `_flush` method not being called, such as due to exception.

Reply:
The current code with the method replacement does not improve this.